src/App/DocumentPyImp.cpp

```diff
@@ -40,6 +40,37 @@
 using namespace App;
 
 
+PyObject*  DocumentPy::addProperty(PyObject *args)
+{
+    char *sType,*sName=nullptr,*sGroup=nullptr,*sDoc=nullptr;
+    short attr=0;
+    std::string sDocStr;
+    PyObject *ro = Py_False, *hd = Py_False;
+    if (!PyArg_ParseTuple(args, "s|ssethO!O!", &sType,&sName,&sGroup,"utf-8",&sDoc,&attr,
+        &PyBool_Type, &ro, &PyBool_Type, &hd))
+        return nullptr;
+
+    if (sDoc) {
+        sDocStr = sDoc;
+        PyMem_Free(sDoc);
+    }
+
+    getDocumentPtr()->addDynamicProperty(sType,sName,sGroup,sDocStr.c_str(),attr,
+            Base::asBoolean(ro), Base::asBoolean(hd));
+
+    return Py::new_reference_to(this);

```

Comment:
@wwmayer Is it sensible to return a reference to the object (Document in this case) rather than to the created property ?
I saw it's same in DocumentObjectPy implementation, so it's a general question. :)

Reply:
We don't have Python wrappers for Property classes. When trying to access a property of a document(object) you get directly its value. So, yes, it's sensible to return a reference of itself.